core/src/main/java/org/testcontainers/containers/wait/HttpWaitStrategy.java

```diff
@@ -21,12 +22,25 @@
      *
      * @param statusCode the expected status code
      * @return this
+     * @deprecated Use {@link #forStatusCodes(Integer...)} instead
      */
+    @Deprecated
```

Comment:
Do you really think we should deprecate the old API? From a technical standpoint you are of course right, I'm just thinking about the developer experience, with a lot of people using this API right now I assume.

Reply:
I would keep this method and add one more with **ranges**. ES' case (two values, 200 and 401) is not as popular as e.g. ranges like 200-299 where any 2xx code is considered valid.

Reply:
Or, to keep this PR small, we can allow calling `forStatusCode` multiple times:
```
 Wait.forHttp("/all").forStatusCode(200).forStatusCode(401)`
```

Reply:
If it works for the ES module, I'd be fine with calling the method multiple times.

Reply:
@bsideup 

> ES' case (two values, 200 and 401) is not as popular as e.g. ranges like 200-299 where any 2xx code is considered valid.

Yeah. I thought about that as well as I saw how fabric8 docker maven plugin offers that as well. But my plan was to come with this later may be with something like `forStatusCodes(String...)`.

> ```java
> Wait.forHttp("/all").forStatusCode(200).forStatusCode(401)
> ```

Yeah. I agree with this proposal. And I can then support as well:

```java
Wait.forHttp("/all").forStatusCode(200, 299).forStatusCode(401)
```

WDYT?

Reply:
`forStatusCode(200, 299)` is too unambiguous :(

I would actually go for
```java
Wait.forHttp("/all").forStatusCodeMatching(it -> it >= 200 && it < 300 || it == 401)
```

Reply:
And also keep `forStatusCode`, plus maybe make it chain-able ( `Wait.forHttp("/all").forStatusCode(200).forStatusCode(401)` )

Reply:
Awesome. Sounds like great ideas. I'll do that